.github/workflows/swift_package_test.yml

```
@@ -76,37 +80,44 @@ jobs:
   windows-build:
     name: Windows (${{ matrix.swift_version }} - windows-2022)
     if: ${{ inputs.enable_windows_checks }}
-    runs-on: windows-2022
+    runs-on: ${{ contains(matrix.swift_version, 'nightly') && 'windows-2019' || 'windows-2022' }}
     strategy:
       fail-fast: false
       matrix:
-        swift_version: ['5.9', '6.0']
+        swift_version: ['5.9', '6.0', 'nightly', 'nightly-6.0']
         exclude:
           - ${{ fromJson(inputs.windows_exclude_swift_versions) }}
     steps:
-      - name: Pull Docker image
-        run: docker pull swift:${{ matrix.swift_version }}-windowsservercore-ltsc2022
       - name: Checkout repository
         uses: actions/checkout@v4
-      - name: Build / Test
-        timeout-minutes: 60
-        run: docker run -v ${{ github.workspace }}:C:\source swift:${{ matrix.swift_version }}-windowsservercore-ltsc2022 cmd /s /c  "swift --version & swift test --version & cd C:\source\ & ${{ inputs.windows_build_command }} ${{  inputs.swift_flags }}"
-
-  windows-nightly-build:
-    name: Windows (${{ matrix.swift_version }} - windows-2019)
-    if: ${{ inputs.enable_windows_checks }}
-    runs-on: windows-2019
-    strategy:
-      fail-fast: false
-      matrix:
-        swift_version: ['nightly', 'nightly-6.0']
-        exclude:
-          - ${{ fromJson(inputs.windows_exclude_swift_versions) }}
-    steps:
       - name: Pull Docker image
-        run: docker pull swiftlang/swift:${{ matrix.swift_version }}-windowsservercore-1809
-      - name: Checkout repository
-        uses: actions/checkout@v4
+        run: |
+          if ("${{ matrix.swift_version }}".Contains("nightly")) {
+            $Image = "swiftlang/swift:${{ matrix.swift_version }}-windowsservercore-1809"
+          } else {
+            $Image = "swift:${{ matrix.swift_version }}-windowsservercore-ltsc2022"
+          }
+          docker pull $Image
```

could we have save the image name as output just to remove duplication below:
```suggestion
          docker pull $Image
          echo "image=$Image" >> "$GITHUB_OUTPUT"
```

Then in the "Create test script" section we can do
```yaml
docker run -v ${{ github.workspace }}:C:\source -v $env:TEMP\test-script:C:\test-script ${{ steps.pull-docker-image.outputs.image }}" C:\test-script\run.cmd
```

(need to set the id of the step to `pull-docker-image`)

See: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-output-parameter

---

Great suggestion. Updated and kicked off another test run: https://github.com/ahoppen/swift-format/actions/runs/11505380578